lib/davinci_dtr_test_kit/client_groups/adaptive_questionnaire/custom/dtr_full_ehr_custom_adaptive_workflow_group.rb

```diff
@@ -0,0 +1,128 @@
+        the `QuestionnaireResponse` as completed.
+
+      At least two answers should be pre-populated across all sets of questions.
+    )
+    run_as_group
+    config(
+      options: { form_type: 'adaptive', next_tag: "custom_#{CLIENT_NEXT_TAG}" },
+      inputs: {
+        custom_questionnaire_package_response: {
+          name: 'adaptive_custom_questionnaire_package_response',
+          title: 'Custom Questionnaire Package Response JSON for Adaptive form',
+          description: %(
+              A JSON PackageBundle may be provided here to replace Inferno's response to
+              the $questionnaire-package request. Note: Ensure that the questionnaire package
+              has an empty Adaptive Questionnaire.
+            )
+        }
+      }
+    )
+
+    input_order :access_token
+
+    group do
+      id :dtr_full_ehr_custom_adaptive_retrieval
+      title 'Retrieving the Adaptive Questionnaire'
+
+      # Test 0: attest to launch
+      test from: :dtr_full_ehr_launch_attest,
+           config: {
+             options: {
+               attestation_message: 'I attest that DTR has been launched in the context of a patient with data that will exercise pre-population logic in the provided static questionnaire resulting in at least 2 pre-populated answers.' # rubocop:disable Layout/LineLength
+             }
+           },
+           title: 'Launch DTR (Attestation)'
+      # Test 1: Recieve questionnaire-package and next-question requests
+      test from: :dtr_full_ehr_custom_adative_request
+      # Test 2: validate the $questionnaire-package request body
+      test from: :dtr_qp_request_validation
+      # Test 3: validate the $next-question requests body
+      test from: :dtr_adaptive_next_question_request_validation
+      # Test 4: validate the QuestionnaireResponse in the input parameter
+      test from: :dtr_adaptive_response_validation do
+        description %(
+          Verify that all submitted QuestionnaireResponse resources meet the following criteria:
+          - Conform to the [SDCQuestionnaireResponseAdapt](http://hl7.org/fhir/uv/sdc/StructureDefinition/sdc-questionnaireresponse-adapt).
+          - Include source extensions indicating whether answers were manually entered,
+            automatically pre-populated, or manually overridden (for `completed` QuestionnaireResponse).
+          - Provide answers for all required items in their contained Questionnaire.
+        )
+        input :custom_next_question_questionnaires
+      end
+      # Test 5: validate the user provided $questionnaire-package response
+      test from: :dtr_custom_qp_validation
+      # Test 6: verify the custom response has the necessary libraries for pre-population
+      test from: :dtr_custom_questionnaire_libraries
+      # Test 7: validate the user provided $next-question questionnaires
+      test from: :dtr_custom_next_questionnaire_validation
```

Comment:
I'm seeing a purple error on this test when no $q-p or $next-question requests are made during the test
<img width="1241" alt="image" src="https://github.com/user-attachments/assets/f450564e-814a-4794-b99a-fcad16a99c8e" />

Reply:
updated, should now skip if no requests are made